Checksec.h

```diff
@@ -114,32 +111,140 @@ class LoadedImage {
+ *
+ * Every mitigation supported by winchecksec is in one of these states.
+ */
+enum class EXPORT MitigationPresence {
+    Present,        /**< The mitigation is present */
+    NotPresent,     /**< The mitigation is not present */
+    NotApplicable,  /**< The mitigation is not applicable on this input */
+    NotImplemented, /**< Support for detecting this mitigation is not implemented */
+};
+
+/**
+ * Represents a "report" on a particular security mitigation.
+ */
+struct EXPORT MitigationReport {
+    /**
+     * A MitigationPresence indicating the mitigation's state.
+     */
+    MitigationPresence presence;
+
+    /**
+     * A brief description of the mitigation.
+     */
+    std::string description;
+
+    /**
+     * An optional explanation of the mitigation's detection (or non-detection).
+     */
+    std::optional<std::string> explanation;
+
+    /**
+     * @return true if `presence` is \ref MitigationPresence::Present, false otherwise
+     */
+    operator bool() const { return presence == MitigationPresence::Present; }
+};
 
+/**
+ * Represents the main winchecksec interface.
+ */
 class EXPORT Checksec {
    public:
     Checksec(std::string filepath);
 
     json toJson() const;
 
+    /**
+     * @return a MitigationReport indicating whether the program can be loaded from a dynamic base
+     *  address (i.e. `/DYNAMICBASE`)
+     */
     const MitigationReport isDynamicBase() const;
+
+    /**
+     * @return a MitigationReport indicating whether the program has **effective** ASLR
+     *  (i.e., has a dynamic base and unstripped relocations, or is in a managed runtime like .NET)
+     */
     const MitigationReport isASLR() const;
+
+    /**
+     * @return a MitigationReport indicating whether the program supports 64-bit ASLR
+     */
     const MitigationReport isHighEntropyVA() const;
+
+    /**
+     * @return a MitigationReport indicating whether the program's integrity
+     *  must be checked at load time
+     */
     const MitigationReport isForceIntegrity() const;
+
+    /**
+     * @return a MitigationReport indicating whether the program supports
+     *  NX, (a.k.a. DEP, `W^X`)
+     */
     const MitigationReport isNX() const;
+
+    /**
+     * @return a MitigationReport indicating whether the operating system
+     *  should attempt a manifest lookup and load for the program
+     */
     const MitigationReport isIsolation() const;
+
+    /**
+     * @return a MitigationReport indicating whether the program uses Structured Exception Handlers
+     */
     const MitigationReport isSEH() const;
+
+    /**
+     * @return a MitigationReport indicating whether the program supports Control Flow Guard
+     */
     const MitigationReport isCFG() const;
+
+    /**
+     * @return a MitigationReport indicating whether the program contains a (partially) valid
+     *  Authenticode signature
+     *
+     * @note See the [`uthenticode`](https://trailofbits.github.io/uthenticode/uthenticode_8h.html)
```

Comment:
Do you want to link to this specific page here? Would the `index.html` page be more appropriate, as it contains more of an overview of what `uthenticode` is, rather than specifically API/code documentation
```suggestion
     * @note See the [`uthenticode`](https://trailofbits.github.io/uthenticode/index.html)
```
I don't have a preference, just checking if that's the page you intended to link

Reply:
Good call!